tools/toolchain/mkmap_data.py

```diff
@@ -0,0 +1,798 @@
+        self.email = email
+        self.mail_events = mail_events
+
+        if self.email is None:
+            self.email = ""
+
+        if self.mail_events is None:
+            self.mail_events = "n"
+
+    def __str__(self):
+        return (
+            str(self.__class__)
+            + "\n"
+            + "\n".join(
+                (str(item) + " = " + str(self.__dict__[item]) for item in self.__dict__)
+            )
+        )
+
+    def write_job_pbs(self, job_fname, cmd):
+        self.job_fname = job_fname
+
+        job_file = open(self.job_fname, "w")
+
+        print("cmd:", cmd)
+
+        self.job_template = (
+            "#!/bin/tcsh\n"
+            "#PBS -N " + self.job_fname + "\n"
+            "#PBS -A " + self.account + "\n"
+            "#PBS -l walltime=" + self.walltime + "\n"
+            "#PBS -q " + self.queue + "\n"
+            "#PBS -j oe" + "\n\n"
+            "#PBS -m " + self.mail_events + "\n"
+            # "#PBS -M " + self.email + "\n"
+            "#PBS -l select="
+            + self.nnodes
+            + ":ncpus="
+            + self.nproc
+            + ":mpiprocs="
+            + self.ppn
+            + ":mem="
+            + self.mem
+            + "\n"
+            "set MPI_SHEPHERD=true" + "\n"
+            "mpiexec_mpt " + cmd + "\n"
+        )
+        job_file.write(self.job_template)
+        job_file.close()
+
+    def schedule(self):
+
+        qsub = subprocess.run(["qsub", self.job_fname], stdout=subprocess.PIPE)
+        qsub_out = qsub.stdout.decode("utf-8")
+
+        print("\n Job successfully submitted:", qsub_out, "\n")
+        job_id = re.findall(r"^\D*(\d+)", qsub_out)
+        self.job_id = "".join(job_id)
+
+    def check_status(self):
+        """
+        Quick function to check the status of the submitted job.
+        """
+
+        cmd = "qstat -x " + self.job_id + "|awk '{print $5}'"
+
+        tmp = subprocess.Popen(
+            cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        out, err = tmp.communicate()
+        out = out.decode("utf-8").split("\n")
+        self.job_status = out[2]
+
+        # qsub = subprocess.run(["qstat","-x", job_id2], stdout=subprocess.PIPE)
+        # qsub_out = qsub.stdout.decode("utf-8").split("\n")
+        # print (qsub_out)
+
+
+class MakeMapper:
+    """
+    A class to encapsulate the necessary information for making
+    a mapping / weight file
+
+    ...
+
+    Attributes
+    ----------
+    src_mesh : MeshType
+        Source mesh in MeshType which contains the information of
+        src mesh file.
+
+    dst_mesh : MeshType
+        Destination mesh in MeshType which contains the information of
+        dst mesh file.
+
+    map_dir : str
+        Directory where you want to store your weight/mapping files.
+
+    mapping_fname: str
+        Mapping /weight filename, if not explicitly given the
+        class will create one based on src_mesh and dst_mesh information.
```

Comment:
This is similar to res and mask, where the advanced user may choose to supply their own value, right? I have the same concern here as I do about res and mask.